### PR TITLE
[1.3] restrict chromium requests

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ps48 — APPROVED** [review]

**Comment by joshuali925** [comment]
something might be wrong with github CI, tests can pass locally but times out here, checking

**Review by davidcui1225 — APPROVED** [review]

**Comment by joshuali925** [comment]
somehow got this error only in github CI, and it times out instead of failing directly. adding try catch avoids it
```js
    Evaluation failed: ReferenceError: cov_90h8ih9c3 is not defined

      209 |   await addReportStyle(page);
      210 |
    > 211 |   const numDisallowedTags = await page.evaluate(
          |                                        ^
```

**Review by vamsi-amazon — APPROVED** [review]

**Review by anirudha — APPROVED** [review]
Lgtm

File: dashboards-reports/server/routes/utils/constants.ts [hunk 1]
```diff
@@ -93,6 +93,8 @@ const ipv6Regex = /(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:)
 const localhostRegex = /localhost:([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])/g;
 const iframeRegex = /iframe/g;
 
+export const ALLOWED_HOSTS = /^(0|0.0.0.0|127.0.0.1|localhost|(.*\.)?(opensearch.org|aws.a2z.com))$/;
```
Comment (dblock) [inline dashboards-reports/server/routes/utils/constants.ts:96]: Why does this have `aws.a2z.com`? This is an Amazon url.

Comment (joshuali925) [reply dashboards-reports/server/routes/utils/constants.ts:96]: replied in https://github.com/opensearch-project/dashboards-reports/pull/425#discussion_r954106923

**Pull request merged**